src/components/Time/Replay.tsx

```diff
@@ -0,0 +1,384 @@
+import * as React from 'react';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import { ThunkDispatch } from 'redux-thunk';
+import { KialiAppState } from 'store/Store';
+import DatePicker from 'react-datepicker';
+import { replayWindowSelector, replayQueryTimeSelector, durationSelector } from 'store/Selectors';
+import { Tooltip, ButtonVariant, Button, Text } from '@patternfly/react-core';
+import { ReplayWindow, DurationInSeconds, IntervalInMilliseconds, TimeInMilliseconds } from 'types/Common';
+import ToolbarDropdown from 'components/ToolbarDropdown/ToolbarDropdown';
+import { UserSettingsActions } from 'actions/UserSettingsActions';
+import { KialiAppAction } from 'actions/KialiAppAction';
+import Slider from 'components/IstioWizards/Slider/Slider';
+import { KialiIcon } from 'config/KialiIcon';
+import { style } from 'typestyle';
+import { toString } from './LocalTime';
+import { DurationDropdownContainer } from 'components/DurationDropdown/DurationDropdown';
+import { serverConfig } from 'config';
+import { PfColors } from 'components/Pf/PfColors';
+
+type ReduxProps = {
+  duration: DurationInSeconds;
+  replayQueryTime: TimeInMilliseconds;
+  replayWindow: ReplayWindow;
+
+  setReplayQueryTime: (replayQueryTime: TimeInMilliseconds) => void;
+  setReplayWindow: (replayWindow: ReplayWindow) => void;
+  toggleReplayActive: () => void;
+};
+
+type ReplayProps = ReduxProps & {
+  id: string;
+};
+
+type ReplayState = {
+  isReplaying: boolean;
+  refresherRef?: number;
+  replayFrame: number;
+  replayFrameCount: number;
+  replaySpeed: IntervalInMilliseconds;
+};
+
+type ReplaySpeed = {
+  speed: IntervalInMilliseconds;
+  text: string;
+};
+
+export const ReplayColor = PfColors.LightBlue200;
+
+// key represents replay interval in seconds
+const replayIntervals = {
+  60000: '1 minute',
+  300000: '5 minutes',
+  600000: '10 minutes',
+  1800000: '30 minutes'
+};
+
+// key represents refresh interval in seconds
```

Comment:
"in seconds"? Isn't it miliseconds instead?
Can you elaborate on how speed is used?

Reply:
Stale comment, I originally had this in seconds but converted to ms when consolidating on ms.  Because this is a replay we don't have to, and really don't want to, refresh in real-time.  It's too tedious to wait.  So "speed" is basically how fast the replay moves.  It  maps to a refresh rate but is a more simple interface than explicitly setting a number of seconds.